src/consolelogger.c

```diff
@@ -7,12 +7,194 @@
 #include "azure_c_shared_utility/xlogging.h"
 #include "azure_c_shared_utility/consolelogger.h"
 
+#ifdef _MSC_VER
+#include "windows.h"
```

Comment:
Can we move this to be only around the getlastError() function.  I may want to use the vprintf_alloc function in linux.